app/src/ui/changes/filter-changes-list.tsx

```diff
@@ -1149,39 +1197,254 @@ export class FilterChangesList extends React.Component<
+          <Button
+            className="filter-button remove-filter"
+            onClick={this.clearNonTextFilters}
+            ariaLabel="Clear 1 additional filter(s)"
+            tooltip="Clear 1 additional filter(s)"
+          >
+            <Octicon symbol={octicons.x} />
+          </Button>
+        ) : null}
+
+        <span>
+          {this.renderBetaPill()}
+
+          <Button
+            className={classNames('filter-button', {
+              active: this.state.filterToIncludedCommit,
+            })}
+            onClick={this.openFilterOptions}
+            ariaExpanded={this.state.isFilterOptionsOpen}
+            onButtonRef={this.onFilterOptionsButtonRef}
+            tooltip={
+              !this.props.canFilterChanges
+                ? 'Enable Filter (Beta)'
+                : 'Filter Options'
+            }
+            ariaLabel={
+              !this.props.canFilterChanges
+                ? 'Enable Filter (Beta)'
+                : 'Filter Options'
+            }
+          >
+            <Octicon symbol={octicons.filter} />
+            {this.props.canFilterChanges ? (
+              <Octicon symbol={octicons.triangleDown} />
+            ) : (
+              <Octicon symbol={octicons.triangleUp} />
+            )}
+          </Button>
+          {this.state.isFilterOptionsOpen && this.renderFilterOptions()}
+        </span>
       </div>
     )
   }
 
+  private renderBetaPill = () => {
+    if (this.props.canFilterChanges) {
+      return
+    }
+
+    return (
+      <span className="beta-pill">
+        <span>
+          <span>Beta</span>
+        </span>
+      </span>
+    )
+  }
+
+  private onFilterOptionsButtonRef = (buttonRef: HTMLButtonElement | null) => {
+    this.filterOptionsButtonRef = buttonRef
+  }
+
+  private onFilterWarningButtonRef = (buttonRef: HTMLButtonElement | null) => {
+    this.filterWarningButtonRef = buttonRef
+  }
+
+  private openFilterWarning = () => {
+    this.setState({ isFilesWarningOpen: true })
+  }
+
+  private closeFilterWarning = () => {
+    this.setState({ isFilesWarningOpen: false })
+  }
+
+  private renderFilterWarning() {
+    return (
+      <Popover
+        className="filter-popover"
+        ariaLabelledby="filter-warning-header"
+        anchor={this.filterWarningButtonRef}
+        anchorPosition={PopoverAnchorPosition.BottomRight}
+        decoration={PopoverDecoration.Balloon}
+        onMousedownOutside={this.closeFilterWarning}
+        onClickOutside={this.closeFilterWarning}
+      >
+        <div className="filter-popover-header">
+          <h3 id="filter-warning-header">Committing Hidden Files</h3>
+          <button
+            className="close"
+            onClick={this.closeFilterWarning}
+            aria-label="Close"
+          >
+            <Octicon symbol={octicons.x} />
+          </button>
+        </div>
+        <div className="filter-warning-text">
+          You have a filter applied. There are{' '}
```

Comment:
Does "filter" refer to what's in the textbox, the filter option, or both?

Reply:
I used the phrase "a filter" to keep it generic for if we had more filters later. But at this point the only one that makes this happen is the text filter